HsUtils.hs

```diff
@@ -41,6 +41,42 @@ isOp (UnQual _ Symbol{}) = True
 isOp (Special _ Cons{})  = True
 isOp _                   = False
 
+isSpecial :: QName () -> Bool
+isSpecial (Special _ _)  = True
+isSpecial _              = False
+
+unQual :: QName () -> Name ()
+unQual (UnQual _ n) = n
+unQual (Qual _ _ n) = n
+unQual (Special _ _)  = error "Unexpected special con"
+
+definedName :: Match l -> Name l
+definedName (Match _ f _ _ _) = f
+definedName (InfixMatch _ _ f _ _ _) = f
+
+replaceName :: Data a => Name () -> Name () -> a -> a
+replaceName pre post = everywhere (mkT go `extT` go')
+  where
+    go :: QName () -> QName ()
+    go n | isSpecial n = n
```

Comment:
`isSpecial n` case not needed?

Reply:
If we didn't have this check, the second guard would cause an error (c.f. the `Special` case of `unQual`).

Reply:
ah, got it, thanks